src/NLog/LogFactory.cs

```diff
@@ -1166,6 +1168,19 @@ public LogFactory LoadConfiguration(string configFile)
             return LoadConfiguration(configFile, optional: false);
         }
 
+        /// <summary>
+        /// Reload the current logging configuration and apply it.
+        /// </summary>
+        /// <returns>LogFactory instance for fluent interface</returns>
+        public LogFactory ReloadConfiguration()
+        {
+            InternalLogger.Info("Reloading configuration...");
+            Configuration = Configuration?.ReloadNewConfig();
+
+            NotifyConfigurationReloaded();
```

Comment:
Why not try-catch and call `NotifyConfigurationReloaded` with the Exception ?

Reply:
Should the Reload not fail when Configuration == null ? (Call NotifyConfigurationReloaded with exception)

Reply:
Instead of calling `Configuration?.ReloadNewConfig()` then check if `_config == null`. This means one should just call Configuration-property and it will automatically load the available NLog.config.

Reply:
Partly done,

> Should the Reload not fail when Configuration == null ? (Call NotifyConfigurationReloaded with exception)

doubting about this.

Reply:
Then at least not call `NotifyConfigurationReloaded` when no reload is performed

Reply:
I'm doubting about this. We invoked a reload, but nothing has changed. If we have to listen for changes, then we have ConfigurationChanged event?